web/src/utils/iframes.ts

```diff
@@ -0,0 +1,190 @@
+  element.appendChild(span)
+  span.appendChild(document.createTextNode(source))
+  element.appendChild(document.createElement('br'))
+}
+
+export const showMessage = (text: string, element: HTMLDivElement, iframeSource?: string): void => {
+  const textNode = document.createTextNode(text)
+  if (iframeSource) {
+    showSource(element, iframeSource)
+  }
+  element.appendChild(textNode)
+}
+
+const showOptIn = (
+  text: string,
+  iframeContainer: HTMLDivElement,
+  source: string,
+  updateCookie: (source: string) => void,
+  index: number,
+): void => {
+  const onClickHandler = () => {
+    updateCookie(source)
+  }
+
+  const className = `iframe-info-text`
+  const elementId = `${className}${source}${index}`
+  const container = getContainer(iframeContainer, className, elementId)
+  if (!container) {
+    return
+  }
+
+  const id = `opt-in-checkbox-${source}${index}`
+  const checkbox = document.createElement('input')
+  checkbox.type = 'checkbox'
+  checkbox.name = 'opt-in-checkbox'
+  checkbox.id = id
+  checkbox.onclick = onClickHandler
+  const label = document.createElement('label')
+  showSource(label, source)
+  label.htmlFor = id
+  label.appendChild(document.createTextNode(text))
+  container.appendChild(label)
+  container.appendChild(checkbox)
+}
+
+const showSettingsButton = (element: HTMLDivElement, t: TFunction): void => {
+  const link = document.createElement('a')
+  link.innerHTML = t('layout:settings')
+  link.id = 'opt-in-settings-link'
+  link.href = `/${CONSENT_ROUTE}`
+  element.appendChild(link)
+}
+
+const showMessageWithSettings = (
+  text: string,
+  iframeContainer: HTMLDivElement,
+  t: TFunction,
+  source: string,
+  iframeIndex: number,
+  removeOptIn: boolean,
+) => {
+  if (removeOptIn) {
+    removeOptInContainer(`iframe-info-text${source}${iframeIndex}`)
+  }
+  const className = `iframe-info-text`
+  const elementId = `${className}${source}${iframeIndex}`
+  const container = getContainer(iframeContainer, className, elementId)
+  if (!container) {
+    return
+  }
+  iframeContainer.appendChild(container)
+  showMessage(text, container, source)
+  showSettingsButton(container, t)
+}
+export const handleAllowedIframeSources = (
+  iframe: HTMLIFrameElement,
+  externalSourcePermissions: ExternalSourcePermissions,
+  storedIframeSource: string,
+  t: TFunction,
+  onUpdateCookie: (source: string) => void,
+  iframeIndex: number,
+  supportedSource: string,
+  viewportSmall: boolean,
+  deviceWidth: number,
+): void => {
+  const permission = supportedSource ? externalSourcePermissions[supportedSource] : undefined
+  const iframeId = `iframe-container${supportedSource}${iframeIndex}`
+  const iframeContainer = getIframeContainer(iframeId, viewportSmall, iframe, deviceWidth)
+
+  if (permission === undefined) {
+    const message = `${t('consent:knownResourceOptIn')}`
+    showOptIn(message, iframeContainer, supportedSource, onUpdateCookie, iframeIndex)
+  } else if (permission && storedIframeSource) {
```

Comment:
`storedIframeSource` is never undefined/null
```suggestion
  } else if (permission) {
```